### PR TITLE
Must join a team

### PR DESCRIPTION
**Pull request by ndavison**
This PR enables an agency to require their users to be a member of a team before they can perform actions such as creating drafts, editing drafts, publishing, creating work orders, downloading responses etc.

It does this by adding a new column to the agency table, `must_join_team`, which will be a toggle in the admin portal agency edit form. Once enabled, any user not in a team in the agency (including those who signup) will be prompted with a page explaining to the user they must join a team, where they can request to join an existing team in the agency. This sends an email to the chosen team's leads, who can decline the request with a reason, or add the user to their team (using the existing team editing process).

### CODE REVIEW

File: apps/marketplace/components/Teams/Dashboard/TeamsOverview.js [hunk 1]
```diff
@@ -30,6 +30,10 @@ export class TeamsOverview extends Component {
 
     return (
       <div className={styles.teams}>
+        <div className={styles.optin}>
```
Comment (tobiasbrown) [inline apps/marketplace/components/Teams/Dashboard/TeamsOverview.js:33]: Did you try to use the margin class from `main.scss`?

File: src/bundles/ApplicationsAdmin/components/Agency/Agency.js [hunk 2]
```diff
@@ -43,6 +43,11 @@ class Agency extends React.Component {
     } else {
       data.whitelisted = false
     }
+    if (data.must_join_team === "on") {
```
Comment (tobiasbrown) [inline src/bundles/ApplicationsAdmin/components/Agency/Agency.js:46]: Where is this `on` value coming from?

File: apps/marketplace/components/Teams/Dashboard/TeamsOverview.js (hunk 1 above)
Comment (ndavison) [reply apps/marketplace/components/Teams/Dashboard/TeamsOverview.js:33]: I did think about it but the component was already importing its own styles with margins etc so added to what was already there.

File: src/bundles/ApplicationsAdmin/components/Agency/Agency.js (hunk 2 above)
Comment (ndavison) [reply src/bundles/ApplicationsAdmin/components/Agency/Agency.js:46]: This is the value coming back from the checkbox component (on or off) - it's consistent with the other radios in the same form in the lines above this.

**Pull request merged**